x/projects/keeper/creation.go

```diff
@@ -2,61 +2,124 @@ package keeper
+	"strconv"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/lavanet/lava/utils"
+	plantypes "github.com/lavanet/lava/x/plans/types"
 	"github.com/lavanet/lava/x/projects/types"
 )
 
 // add a default project to a subscription, add the subscription key as
-func (k Keeper) CreateAdminProject(ctx sdk.Context, subscriptionAddress string, totalCU uint64, cuPerEpoch uint64, providers uint64, vrfpk string) error {
-	return k.CreateProject(ctx, subscriptionAddress, types.ADMIN_PROJECT_NAME, subscriptionAddress, true, totalCU, cuPerEpoch, providers, math.MaxUint64, vrfpk)
+func (k Keeper) CreateAdminProject(ctx sdk.Context, subscriptionAddress string, plan plantypes.Plan, vrfpk string) error {
+	projectData := types.ProjectData{
+		Name:        types.ADMIN_PROJECT_NAME,
+		Description: types.ADMIN_PROJECT_DESCRIPTION,
+		Enabled:     true,
+		ProjectKeys: []types.ProjectKey{{Key: subscriptionAddress, Types: []types.ProjectKey_KEY_TYPE{types.ProjectKey_DEVELOPER}, Vrfpk: vrfpk}},
+		Policy:      nil,
+	}
+	return k.CreateProject(ctx, subscriptionAddress, projectData, plan)
 }
 
 // add a new project to the subscription
-func (k Keeper) CreateProject(ctx sdk.Context, subscriptionAddress string, projectName string, adminAddress string, enable bool, totalCU uint64, cuPerEpoch uint64, providers uint64, geolocation uint64, vrfpk string) error {
-	project := types.CreateProject(subscriptionAddress, projectName)
-	var emptyProject types.Project
+func (k Keeper) CreateProject(ctx sdk.Context, subscriptionAddress string, projectData types.ProjectData, plan plantypes.Plan) error {
+	project, err := types.CreateProject(subscriptionAddress, projectData.GetName(), projectData.GetDescription(),
+		projectData.GetEnabled())
+	if err != nil {
+		return err
+	}
 
+	var emptyProject types.Project
 	blockHeight := uint64(ctx.BlockHeight())
 	_, found := k.projectsFS.FindEntry(ctx, project.Index, blockHeight, &emptyProject)
 	// the project with the same name already exists if no error has returned
 	if found {
 		return utils.LavaError(ctx, ctx.Logger(), "CreateEmptyProject_already_exist", map[string]string{"subscription": subscriptionAddress}, "project already exist for the current subscription with the same name")
 	}
 
-	project.Policy.EpochCuLimit = cuPerEpoch
-	project.Policy.TotalCuLimit = totalCU
-	project.Policy.MaxProvidersToPair = providers
-	project.Policy.GeolocationProfile = geolocation
+	policy := projectData.GetPolicy()
+	if policy == nil {
+		projectDefaultPolicy := types.Policy{
+			ChainPolicies:      []types.ChainPolicy{},
+			GeolocationProfile: math.MaxUint64,
+			TotalCuLimit:       plan.PlanPolicy.GetTotalCuLimit(),
+			EpochCuLimit:       plan.PlanPolicy.GetEpochCuLimit(),
+			MaxProvidersToPair: plan.PlanPolicy.GetMaxProvidersToPair(),
+		}
+		policy = &projectDefaultPolicy
+	}
 
-	project.AppendKey(types.ProjectKey{Key: adminAddress, Types: []types.ProjectKey_KEY_TYPE{types.ProjectKey_ADMIN}, Vrfpk: vrfpk})
+	project.AdminPolicy = *policy
```

Comment:
let subscriptionpolicy and adminpolicy to be nil. this is to be able to set them with no limitation except the planpolicy (that cant be nil).
this is so when the plan policy changes the users will not need to manually change the admin/subscription policy in order to get the new policy

Reply:
for the verify of the policy, let the function recieve a list of policies and that way we can change the amount of policies we have to check